src/main/java/net/mcreator/plugin/PluginInfo.java

```diff
@@ -18,14 +18,17 @@
 
 package net.mcreator.plugin;
 
+
 public class PluginInfo {
 
 	private String name;
 	private String description;
 	private String author;
 	private String version;
+	private String dependency;

```

Comment:
Change this to List<String> dependencies so multiple can be defined

Reply:
I already tried to put as an ArrayList, but plugins with a dependency never loaded. I can re-try to see if I'm able to see what i made wrong.

Reply:
Try to resolve this, a single dependency is a bit limiting. Try using List<String> instead. Also try printing contents of this list for testing the source of problem.

Reply:
I will try to resolve it for sure. I already printed the content of the dependencies (from the plugin) and the list I used to store plugin ids, and even if both have the same plugin-id, it doesn't work.

Reply:
The problem is probably here: idList.contains(plugin.getInfo().getDependencies())

You can not check if array list contains array list, you will need to iterate through it and check for each one.

I also suggest you to merge first with master branch and do these checks outside loadPluginsFromFolder, but inside PluginLoader constructor after plugins list is filled.

Reply:
I fixed it and it works (tested I think all cases)